### PR TITLE
Implemented Task and Event Class

### PR DESCRIPTION
**Pull request by JunruL**
Since some decisions have not been made during the meeting, I made the decisions to write the complete code. These decisions are temporary and the structure and functionality of the classes may change after next meeting. Here are some ideas that need to be discussed:
- `Task` stores basic information of a task (name, completed, subtasks). It has a tree structure. 
- `MainTask` is a high level class which we create before any sub task is added. `MainTask` servers as a root of the task tree.
- Based on the implementation of `Task` and `MainTask`, any subtask will only have name, completed status, and subtasks. Subtask could not store the deadline and time needed. So subtasks are more like a check list instead of lists of true tasks. (Need to discuss this in next meeting. Another option is to timeNeeded and deadline both optional and combine Task and MainTask. However, as was discussed  in the meeting, the root task should have timeNeeded variable for it changing into an event, so it should not be optional for root task)
- `Event` and `Task` has some common methods so maybe they could have the same parent class
- Separate one-time events and repeated events may make the code look better (see `line 11` to `line 20` in `Event` class) Or anyone has ideas of how to store the repeat time information?

### CODE REVIEW

File: src/main/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package main.java;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+
+public interface Main {
```
Comment (jenci2114) [inline src/main/java/Main.java:6]: The name of this interface sounds a little ambiguous to me

File: src/main/java/Task.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        this.taskName = newName;
+    }
+
+    public void complete() {
+        this.completed = true;
+    }
+
+    public void undoComplete() {
+        this.completed = false;
+    }
```
Comment (jenci2114) [inline src/main/java/Task.java:31]: I would suggest merging those methods with one setter for `completed`

Comment (JunruL) [reply src/main/java/Task.java:31]: Ok. And I will also rename other setters.

File: src/main/java/Main.java (hunk 1 above)
Comment (jenci2114) [reply src/main/java/Main.java:6]: If the interface is only for one specific class, then I think we would not need it unless there would be other similar classes that also hold root tasks in the future

**Pull request merged**